### PR TITLE
[Scaleway] Add module to manage functions

### PR DESCRIPTION
**Pull request by Lunik**
##### SUMMARY

This PR add new modules to manage Scaleway functions services (https://developers.scaleway.com/).

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

- `scaleway_function`
- `scaleway_function_info`

##### ADDITIONAL INFORMATION

This PR is a part of the original PR #5359 that was too large.

This PR require evolution on Scaleway config in ansible-test : https://github.com/ansible/ansible/pull/79141

### CODE REVIEW

**Comment by felixfontein** [comment]
recheck

**Comment by Lunik** [comment]
ready_for_review

File: plugins/modules/scaleway_function.py [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+    type: int
+
+  function_timeout:
+    description:
+      - The length of time your handler can spend processing a request before being stopped.
+    type: str
+
+  handler:
+    description:
+      - The module-name.export value in your function.
```
Comment (felixfontein) [inline plugins/modules/scaleway_function.py:111]: Not sure whether it fits:
```suggestion
      - The C(module-name.export) value in your function.
```

**Review by felixfontein — APPROVED** [review]
Looks good to me! I have a tiny comment, please decide whether you want to apply it or not (just resolve then).

**Comment by felixfontein** [comment]
@Lunik thanks for contributing this!

**Comment by andrewshadura** [comment]
Thanks a lot, I was about to write this myself, and then I found this PR 🙂 ❤️

**Pull request merged**